Building with CMake never defines HAVE_LANGINFO_CODESET

Buliding with CMake + Ninja never adds `-DHAVE_LANGINFO_CODESET` to compiler options, so locale detection is not complete in such situation.

Notably, ArchLinux now builds flac with CMake + Ninja (https://github.com/archlinux/svntogit-packages/blob/packages/flac/trunk/PKGBUILD), which causes some locale issue when tagging FLAC files. See: https://github.com/svend/cuetools/issues/38#issuecomment-1082859037

@NotTsunami Sorry for the late response. Actually the test fails with the following compiler error:
```
/root/workspace/flac/build/CMakeFiles/CMakeTmp/src.c:3:16: error: initializer element is not constant
    3 |     char* cs = nl_langinfo(CODESET);
      |                ^~~~~~~~~~~
```
It seems that right now cs is a global variable initialized with nl_langinfo. Maybe `char* cs = ...` need to be moved into inside the main function?

> @NotTsunami Sorry for the late response. Actually the test fails with the following compiler error:
> 
> ```
> /root/workspace/flac/build/CMakeFiles/CMakeTmp/src.c:3:16: error: initializer element is not constant
>     3 |     char* cs = nl_langinfo(CODESET);
>       |                ^~~~~~~~~~~
> ```
> 
> It seems that right now cs is a global variable initialized with nl_langinfo. Maybe `char* cs = ...` need to be moved into inside the main function?

I'll push another patchset tonight.

> @NotTsunami Sorry for the late response. Actually the test fails with the following compiler error:
> 
> ```
> /root/workspace/flac/build/CMakeFiles/CMakeTmp/src.c:3:16: error: initializer element is not constant
>     3 |     char* cs = nl_langinfo(CODESET);
>       |                ^~~~~~~~~~~
> ```
> 
> It seems that right now cs is a global variable initialized with nl_langinfo. Maybe `char* cs = ...` need to be moved into inside the main function?

Just pushed the new patchset if you want to give it a whirl! I'll try to spin up an Arch box in the next couple days if I don't hear back and attempt to replicate.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -79,6 +79,14 @@ check_function_exists(fseeko HAVE_FSEEKO)
 
 check_c_source_compiles("int main() { return __builtin_bswap16 (0) ; }" HAVE_BSWAP16)
 check_c_source_compiles("int main() { return __builtin_bswap32 (0) ; }" HAVE_BSWAP32)
+check_c_source_compiles("
+    #include <langinfo.h>
+    int main()
+    {
+        char* cs = nl_langinfo(CODESET);
+        return !cs;
+    }"
+    HAVE_LANGINFO_CODESET)
 
 test_big_endian(CPU_IS_BIG_ENDIAN)
 

